### ISSUE TITLE
Skip_when_empty: Unobvious serialization behaviour | update docs

### ISSUE THREAD

**Issue by Khairullin** [schmittjoh/serializer#1314]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | no
| Doc updated? | yes
| Unobvious behaviour? | yes

## Steps required to reproduce the problem

1. Create model class at least with one array<Type> field, initialize it with [], just when declaring field.
2. Write yaml config for model class, with xml_list option for array field
3. Post some test data and try to serialize it into xml format.
4. xml-tag for empty array abscent, it was skipped by default.
5. Try to serialize to json-format. json-field with empty array is present

## Expected Result

Same behaviour for json and xml serialization for same yaml config. "skip_when_empty" option mentioned in docs as "allowed inside  xml_list"

## Actual Result

Serialization behaviour differs. No info in docs about "skip_when_empty" inside "xml_list", even in examples.

**Comment by goetas** [comment]
you are right "skip_when_empty" works well only for json. if you are interested to make it work for php, it would be great
